### ISSUE TITLE
Any plan to support using proxy?

### ISSUE THREAD

**Issue by hurui200320** [martinber/noaa-apt#30]
Hi,
In some area(i.e. China), the network access was limited, and normally I'll use a socks5 as system proxy and a lot of softwares are happy with that. But for this app, it seems like neither it use system proxy nor use a specific one. So any idea of using a proxy?

```
2020-10-04 16:29:10,033 WARN  [noaa_apt::misc] Unable to read cached TLE timestamp: Could not read metadata from input file: 系统找不到指定的文件。 (os error 2). Downloading and caching new TLE
2020-10-04 16:29:10,041 ERROR [noaa_apt] error sending request for url (https://www.celestrak.com/NORAD/elements/weather.txt): error trying to connect: unexpected EOF during handshake
```

Although download and specify that `weather.txt` to the program works OK.

Thanks.

**Comment by martinber** [comment]
Thanks for the suggestion. I thought that most proxies are normally configured system-wide and that applications use them transparently.

Do you normally configure proxies in every application you use? (the browser, spotify, thunderbird, etc). Or the rest of applications just work?

I've only checked [this](https://github.com/iamsimakov/rust-reqwest-socks5/blob/master/src/main.rs), but it requires manual configuration, I should investigate further

**Comment by hurui200320** [comment]
Normally I have a system wide proxy of Socks5, and most app works fine. Some may need to manually configure proxies(Chrome, git etc.). And rest of them I have to create a HTTP proxy to make them happy. 

Looks like that rust lib [has a solution on this](https://github.com/seanmonstar/reqwest/pull/1005).

Also I dig my proxy's log. When the app made a request, I got a warning(using http proxy):
```
2020/10/05 09:38:04 [Warning] [2360297570] v2ray.com/core/app/proxyman/inbound: connection ends > v2ray.com/core/proxy/http: failed to read http request > malformed HTTP request "\x16\x03\x03\x00\x94\x01\x00\x00\x90\x03\x03_zx\xfc\x84k\xe5\xfb\x8a\xdbֱ\xdf\xeb\x03/\x95Z\xa0F\xb5M\x869/I\xa3\u05ce\xe1\xa4\xe7\x00\x00*\xc0,\xc0+\xc00\xc0/\x00\x9f\x00\x9e\xc0$\xc0#\xc0(\xc0'\xc0"
```
or (using socks5):
```
2020/10/05 09:39:52 [Warning] [1127933764] v2ray.com/core/app/proxyman/inbound: connection ends > v2ray.com/core/proxy/http: failed to read http request > malformed HTTP request "\x16\x03\x03\x00\x94\x01\x00\x00\x90\x03\x03_zyh"
```

So it does make request through proxy, but seems like the protocol doesn't match. I notice that [v0.10.8 of reqwest Fix Windows system proxy detection of Fiddler proxies](https://github.com/seanmonstar/reqwest/releases/tag/v0.10.8). And this app are currently using v0.10.4, so I think a version upgrade might sovle this?

**Comment by hurui200320** [comment]
UPDATE: By changing the system proxy addr from `127.0.0.1:1081` to `http://127.0.0.1:1081`, everything works ok.

**Comment by martinber** [comment]
Thanks! So I guess I don't have to do anything?

**Comment by hurui200320** [comment]
Maybe considering update the reqwest to v0.10.8? The changelog said it Fix Windows system proxy detection of Fiddler proxies. So it might add `http://` automatically.

**Comment by martinber** [comment]
I've just released noaa-apt with reqwest v0.10.8, so if you can, check it please

**Comment by hurui200320** [comment]
Sorry for no reply, I managed to get my paper done before ddl. Anyway, with the latest version, everything looks fine. Thanks.
